### PR TITLE
Prl 3014 changes non molestation

### PR DESCRIPTION
**Pull request by khurshedk**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/constants/PrlAppsConstants.java [hunk 1]
```diff
@@ -128,6 +128,9 @@ public class PrlAppsConstants {
     public static final List<String> ROLES = List.of("caseworker-privatelaw-courtadmin",
                                                      "caseworker-privatelaw-judge",
                                                      "caseworker-privatelaw-la");
+
+    public static final List<String> ROLES_JUDGE = List.of("caseworker-privatelaw-judge",
+                                                     "caseworker-privatelaw-la");
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/constants/PrlAppsConstants.java:133]: move them to ENUM. Use manage order PR for this.  Speak to Sonali

File: src/main/java/uk/gov/hmcts/reform/prl/constants/PrlAppsConstants.java [hunk 2]
```diff
@@ -328,5 +331,7 @@ public class PrlAppsConstants {
     public static final String ADJOURNED = "ADJOURNED";
     public static final String POSTPONED = "POSTPONED";
     public static final String COMPLETED = "COMPLETED";
+    public static final String JUDGE = "Judge";
+    public static final String CASEWORKER = "CaseWorker";
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/constants/PrlAppsConstants.java:335]: Move it to ENUM, there is already one.  Speak to Sonali

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/C100IssueCaseController.java [hunk 3]
```diff
@@ -55,30 +59,41 @@ public AboutToStartOrSubmitCallbackResponse issueAndSendToLocalCourt(
...
         if (null != caseData.getCourtList() && null != caseData.getCourtList().getValue()) {
-            String baseLocationId = caseData.getCourtList().getValue().getCode();
+            String[] idEmail = caseData.getCourtList().getValue().getCode().split(":");
+            String baseLocationId = Arrays.stream(idEmail).toArray()[0].toString();
             String key = locationRefDataService.getCourtDetailsFromEpimmsId(baseLocationId, authorisation);
             String[] venueDetails = key.split("-");
             String regionId = Arrays.stream(venueDetails).toArray()[1].toString();
             String courtName = Arrays.stream(venueDetails).toArray()[2].toString();
             String regionName = Arrays.stream(venueDetails).toArray()[4].toString();
             String baseLocationName = Arrays.stream(venueDetails).toArray()[5].toString();
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/C100IssueCaseController.java:73]: move it to a static utility method. use the manage order base branch for this. Speak to Sonali

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/C100IssueCaseController.java [hunk 4]
```diff
@@ -55,30 +59,41 @@ public AboutToStartOrSubmitCallbackResponse issueAndSendToLocalCourt(
...
         }
-
+        caseData.setIssueDate();
         // Generate All Docs and set to casedataupdated.
         caseDataUpdated.putAll(documentGenService.generateDocuments(authorisation, caseData));
-
         // Refreshing the page in the same event. Hence no external event call needed.
         // Getting the tab fields and add it to the casedetails..
         Map<String, Object> allTabsFields = allTabsService.getAllTabsFields(caseData);
+        caseDataUpdated.put("issueDate", LocalDate.now());
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/C100IssueCaseController.java:90]: there is a reason for the old code. its about return and resubmit. is this change valid?

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java [hunk 5]
```diff
@@ -391,15 +392,29 @@ public AboutToStartOrSubmitCallbackResponse sendEmailNotificationOnCaseWithdraw(
...
+            Optional<List<Element<LocalCourtAdminEmail>>> localCourtAdmin = ofNullable(caseData.getLocalCourtAdmin());
+            if (localCourtAdmin.isPresent()) {
+                Optional<LocalCourtAdminEmail> localCourtAdminEmail = localCourtAdmin.get().stream().map(Element::getValue)
+                    .findFirst();
+                if (localCourtAdminEmail.isPresent()) {
+                    String email = localCourtAdminEmail.get().getEmail();
+                    log.info(" Sending notification to court email : {}", email);
+                    caseWorkerEmailService.sendWithdrawApplicationEmailToLocalCourt(caseDetails, email);
+                }
+            }
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java:404]: Is this relevant? work allocation is supposed to take care of this?

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java [hunk 6]
```diff
@@ -391,15 +392,29 @@ public AboutToStartOrSubmitCallbackResponse sendEmailNotificationOnCaseWithdraw(
...
         } else {
             solicitorEmailService.sendWithDrawEmailToFl401SolicitorAfterIssuedState(caseDetails, userDetails);
+            caseWorkerEmailService.sendWithdrawApplicationEmailToLocalCourt(
+                caseDetails,
+                caseData.getCourtEmailAddress());
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java:409]: Is this relevant? work allocation is supposed to take care of this?

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java [hunk 7]
```diff
@@ -391,15 +392,29 @@ public AboutToStartOrSubmitCallbackResponse sendEmailNotificationOnCaseWithdraw(
...
         }
     }
 
     public void sendC100CaseWithDrawEmails(CaseData caseData, CaseDetails caseDetails, UserDetails userDetails) {
         Optional<List<Element<LocalCourtAdminEmail>>> localCourtAdmin = ofNullable(caseData.getLocalCourtAdmin());
         if (localCourtAdmin.isPresent()) {
             String email = localCourtAdmin.get().get(0).getValue().getEmail();
+            log.info(" Sending notification to court email : {}", email);
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java:417]: remove email logger

**Comment by khurshedk** [comment]
Closing this no longer required created new pr for these chnages

**Pull request closed without merging**